src/config.rs

```diff
@@ -482,6 +491,15 @@ impl<Scope: private::ConfigScope> ConfigBuilder<Scope> {
         self
     }
 
+    /// If we should record a history of every redirect location,
+    /// including the request and final locations.
+    ///
+    /// Defaults to false
+    pub fn redirect_history(mut self, v: bool) -> Self {
```

Comment:
Explain that enabling this comes at the cost of allocating/retaining the `Uri` for every redirect hop.